util/hint/hint_processor.go

```diff
@@ -55,22 +61,57 @@ func (hs *HintsSet) ContainTableHint(hint string) bool {
 }
 
 // ExtractTableHintsFromStmtNode extracts table hints from this node.
-func ExtractTableHintsFromStmtNode(node ast.Node) []*ast.TableOptimizerHint {
+func ExtractTableHintsFromStmtNode(node ast.Node, sctx sessionctx.Context) []*ast.TableOptimizerHint {
 	switch x := node.(type) {
 	case *ast.SelectStmt:
 		return x.TableHints
 	case *ast.UpdateStmt:
 		return x.TableHints
 	case *ast.DeleteStmt:
 		return x.TableHints
-	// TODO: support hint for InsertStmt
+	case *ast.InsertStmt:
+		//check duplicated hints
+		checkInsertStmtHintDuplicated(node, sctx)
+		return x.TableHints
 	case *ast.ExplainStmt:
-		return ExtractTableHintsFromStmtNode(x.Stmt)
+		return ExtractTableHintsFromStmtNode(x.Stmt, sctx)
 	default:
 		return nil
 	}
 }
 
+// checkInsertStmtHintDuplicated check whether existed the duplicated hints in both insertStmt and its selectStmt.
+// If existed, it would send a warning message.
+func checkInsertStmtHintDuplicated(node ast.Node, sctx sessionctx.Context) {
+	switch x := node.(type) {
+	case *ast.InsertStmt:
+		if len(x.TableHints) > 0 {
+			var supportedHint *ast.TableOptimizerHint
+			for _, hint := range x.TableHints {
+				if _, ok := supportedHintNameForInsertStmt[hint.HintName.L]; ok {
+					supportedHint = hint
+					break
+				}
+			}
+			if supportedHint != nil {
+				var duplicatedHint *ast.TableOptimizerHint
+				for _, hint := range ExtractTableHintsFromStmtNode(x.Select, nil) {
+					if hint.HintName.L == supportedHint.HintName.L {
+						duplicatedHint = hint
+						break
+					}
+				}
+				if duplicatedHint != nil {
+					warning := fmt.Sprintf("Hint %s(`%v`) is ignored as conflicting/duplicated", duplicatedHint.HintName.O, duplicatedHint.HintData)
```

Comment:
https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html#error_er_warn_conflicting_hint
It's better to define this error message in `tidb/errno` package, and use an error variable here.

Reply:
updated